src/network/websocket/WebSocketServer.cpp

```diff
@@ -190,12 +197,13 @@ void SingleClientWSServer::onMessage(connection_hdl hdl, message_t message) {
 	auto conn = server.get_con_from_hdl(hdl);
 	std::string path = conn->get_resource();
 
-	auto it = protocolMap.find(path);
-	if (it != protocolMap.end()) {
+	auto protocolDataOpt = this->getProtocol(path);
+	if (protocolDataOpt.has_value()) {
+		ProtocolData& pd = protocolDataOpt.value();
 		std::string jsonStr = message->get_payload();
 		LOG_F(1, "Message on %s: %s", path.c_str(), jsonStr.c_str());
 		json obj = json::parse(jsonStr);
-		it->second.protocol->processMessage(obj);
+		pd.protocol->processMessage(obj);
```

Comment:
Can you add a comment here saying that locking isn't necessary? Just so we don't in the future wonder why we didn't add a lock here.